packages/thirdweb/src/react/web/utils/usePreloadWalletProviders.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect } from "react";
+import type { ThirdwebClient } from "../../../client/client.js";
+import { COINBASE } from "../../../wallets/constants.js";
+import type { Wallet } from "../../../wallets/interfaces/wallet.js";
+
+export const usePreloadWalletProviders = ({
+  client,
+  wallets,
+}: { client: ThirdwebClient; wallets: Wallet[] }) => {
+  useEffect(() => {
```

Comment:
this will potentially fire multiple times, need to use `useQuery` here - I'll submit a pr on top

Reply:
addressed in #3331